ratelimits/limit_test.go

```diff
@@ -0,0 +1,333 @@
+	test.AssertError(t, err, "multiple override limits, third entry has bad Id value")
+	test.Assert(t, !os.IsNotExist(err), "test file should exist")
+}
+
+func Test_loadAndParseDefaultLimits(t *testing.T) {
+	newRestistrationsPerIPv4AddressEnumStr := nameToEnumString(NewRegistrationsPerIPAddress)
+	newRegistrationsPerIPv6RangeEnumStr := nameToEnumString(NewRegistrationsPerIPv6Range)
+
+	// Load a single valid default limit.
+	l, err := loadAndParseDefaultLimits("testdata/working_default.yml")
+	test.AssertNotError(t, err, "valid single default limit")
+	test.AssertEquals(t, l[newRestistrationsPerIPv4AddressEnumStr].Burst, int64(20))
+	test.AssertEquals(t, l[newRestistrationsPerIPv4AddressEnumStr].Count, int64(20))
+	test.AssertEquals(t, l[newRestistrationsPerIPv4AddressEnumStr].Period.Duration, time.Second)
+
+	// Load multiple valid default limits.
+	l, err = loadAndParseDefaultLimits("testdata/working_defaults.yml")
+	test.AssertNotError(t, err, "multiple valid default limits")
+	test.AssertEquals(t, l[newRestistrationsPerIPv4AddressEnumStr].Burst, int64(20))
+	test.AssertEquals(t, l[newRestistrationsPerIPv4AddressEnumStr].Count, int64(20))
+	test.AssertEquals(t, l[newRestistrationsPerIPv4AddressEnumStr].Period.Duration, time.Second)
+	test.AssertEquals(t, l[newRegistrationsPerIPv6RangeEnumStr].Burst, int64(30))
+	test.AssertEquals(t, l[newRegistrationsPerIPv6RangeEnumStr].Count, int64(30))
+	test.AssertEquals(t, l[newRegistrationsPerIPv6RangeEnumStr].Period.Duration, time.Second*2)
+
+	// Path is empty string.
+	_, err = loadAndParseDefaultLimits("")
+	test.AssertError(t, err, "path is empty string")
+	test.Assert(t, os.IsNotExist(err), "path is empty string")
+
+	// Path to file which does not exist.
+	_, err = loadAndParseDefaultLimits("testdata/file_does_not_exist.yml")
+	test.AssertError(t, err, "a file that does not exist")
+	test.Assert(t, os.IsNotExist(err), "test file should not exist")
+
+	// Burst cannot be 0.
+	_, err = loadAndParseDefaultLimits("testdata/busted_default_burst_0.yml")
+	test.AssertError(t, err, "single default limit with burst=0")
+	test.Assert(t, !os.IsNotExist(err), "test file should exist")
+
+	// Name cannot be empty.
+	_, err = loadAndParseDefaultLimits("testdata/busted_default_empty_name.yml")
+	test.AssertError(t, err, "single default limit with empty name")
+	test.Assert(t, !os.IsNotExist(err), "test file should exist")
+
+	// Name must be a string representation of a valid Name enumeration.
+	_, err = loadAndParseDefaultLimits("testdata/busted_default_invalid_name.yml")
+	test.AssertError(t, err, "single default limit with invalid name")
+	test.Assert(t, !os.IsNotExist(err), "test file should exist")
+
+	// Multiple entries, second entry has a bad name.
+	_, err = loadAndParseDefaultLimits("testdata/busted_defaults_second_entry_bad_name.yml")
+	test.AssertError(t, err, "multiple default limits, one is bad")
+	test.Assert(t, !os.IsNotExist(err), "test file should exist")
```

Comment:
There should be a test for an invalid period.
```
$ cat ratelimits/testdata/busted_period_0s.yml 
NewRegistrationsPerIPAddress:
  burst: 20
  count: 20
  period: 0s
```

```
	// Invalid period.
	_, err = loadAndParseDefaultLimits("testdata/busted_period_0s.yml")
	test.Assert(t, !os.IsNotExist(err), "test file should exist")
	test.AssertError(t, err, "period must be >= 1s")
```

Reply:
`Test_validateLimit` covers this.